### PR TITLE
Add Queueing and Scheduling

### PR DESCRIPTION
**Pull request by htdvisser**

### CODE REVIEW

File: queue/base.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+// Base interface for Queue
+type Base interface {
+	// Next item in the Queue, this function blocks until the next item is available.
+	// It returns <nil> to all callers when Clean() is called.
+	Next() interface{}
+
+	IsEmpty() bool
+
+	// Clean up the queue
+	Clean()
```
Comment (romeovs) [inline queue/base.go]: `Clear` would make more sense.

File: queue/base.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+// Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+
+// Package queue implements different kinds of queues
+package queue
+
+// Base interface for Queue
+type Base interface {
+	// Next item in the Queue, this function blocks until the next item is available.
+	// It returns <nil> to all callers when Clean() is called.
+	Next() interface{}
```
Comment (romeovs) [inline queue/base.go]: Isn't the standard name for this `Pop` or `Unqueue`, or even `Take`?

File: queue/jit.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+type jitQueue struct {
+	nextMu sync.Mutex
+
+	mu    sync.Mutex
+	queue jitSlice
+
+	changed chan struct{}
+}
+
+// NewJIT returns a new JIT Queue
```
Comment (romeovs) [inline queue/jit.go]: Explain the characteristics of the JIT queue.

File: queue/base.go (hunk 2 above)
Comment (htdvisser) [inline queue/base.go]: Yes, except that those are not supposed to block, which `Next()` does.

**Pull request merged**